### ISSUE TITLE
Change anaconda environment to pip install for python2

### ISSUE THREAD

**Issue by ArDrambyan** [CodeSignal/dockerfiles#153]
We shoud try to change anaconda environment with pip installs to see if it is possible and if it gives us better performance

**Comment by ArDrambyan** [comment]
@menewman @eduard-piliposyan I created python2 image which supports all our libs (except `tensorflow`, I have some problems with installing `tensorflow` right now ) which are installed with pip instead fo `conda`, and I get pretty interesting results.
1. compressed image size in `dockerhub` is 1.2GB
2. execution time with docker pull without including any lib is 400-500ms (without docker pool ~1s), when I include all libs execution time is ~3-3.4s.

So I think that switching from `conda` to pip install is a better option than keeping `conda` and splitting images. The last question that I have is, does it make sense to split python2 and python3 into different images? They are independent and we won't change python2 stuff too much in the future.

**Comment by menewman** [comment]
It seems like the inclusion/exclusion of `tensorflow` could be an important difference here, since (I assume) that is one of the big data science libraries?

@ArDrambyan - Could you help clarify whether this is actually a performance improvement? Before, you said with conda and all libraries, the execution time was 3.5 to 4 seconds without Docker pool:
https://github.com/CodeSignal/dockerfiles/issues/150#issuecomment-633563609

And here, with pip and most libraries (except tensorflow), the execution time is 3 - 3.4 seconds. I'm not sure if you meant with or without Docker pool here, which makes a bit of a difference to how to interpret the results.

Based on that, it sounds like conda vs. pip doesn't make as much of a difference as just whether all libraries are installed? Maybe I'm misunderstanding the results, though?

> The last question that I have is, does it make sense to split python2 and python3 into different images? They are independent and we won't change python2 stuff too much in the future.

I'm generally okay with this, esp. if we see that it can improve runtimes for each language, and esp. because Python 2 is now EOL and shouldn't need to be updated as frequently in the future (except to add libs as needed). Would be curious to see how this affects the image size and the execution time, of course.

**Comment by ArDrambyan** [comment]
When we are using conda, execution time is always 3.5-4 seconds, even when we don't include any lib.

**Comment by ArDrambyan** [comment]
I installed `tensorflow` and the size of the compressed image in the `DockerHub` is 2GB. Also I'm using this code to check that all libs are installed successfully:
```
import pathlib 
import numpy 
import scipy 
import pandas 
import sklearn  
import nltk 
import requests 
import configparser 
import mysql.connector
import mysql 
import sqlalchemy 
import psycopg2 
import toolz 
import appdirs 
import attr
import click 
import toml 
import spacy 
import matplotlib 
import seaborn 
import pytest 
import pymongo 
import flask 
import flask_migrate 
import ipywidgets 
import pandas_profiling
import plotly 
import tweepy 
import eli5 
import pdpbox 
import shap 
import xgboost
import basilica
import tensorflow
import tensorflow.keras

nlp = spacy.load("en_core_web_sm")

print 'Hello!'
```
@menewman I'm going forward and creating an issue for splitting python's image into 2 images and replacing conda installations with pip install

**Comment by menewman** [comment]
@ArDrambyan - Okay, sounds very promising! Exciting that the size is down even with all libraries installed. I think the part of this that I found confusing is the following:

> When we are using conda, execution time is always 3.5-4 seconds, even when we don't include any lib.

In this comment...
https://github.com/CodeSignal/dockerfiles/issues/150#issuecomment-633563609

> using codesignal/python:v5.6 (without new libs), it takes 0.5-0.7s with docker pool and 1-1.5s without docker pool

I was assuming your benchmarking in the prior issue was done with conda. But if it wasn't, perhaps that's the part of it that I misunderstood!

**Comment by ArDrambyan** [comment]
Another good news, we are going to support python3.8 with all libs
